eureka2-client/src/main/java/com/netflix/eureka2/client/functions/ChangeNotificationFunctions.java

```diff
@@ -0,0 +1,88 @@
+package com.netflix.eureka2.client.functions;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.netflix.eureka2.interests.ChangeNotification;
+import com.netflix.eureka2.interests.ChangeNotification.Kind;
+import com.netflix.eureka2.utils.rx.RxFunctions;
+import rx.Observable;
+import rx.Observable.Transformer;
+import rx.functions.Func1;
+
+/**
+ * @author Tomasz Bak
+ */
+public final class ChangeNotificationFunctions {
+
+    private ChangeNotificationFunctions() {
+    }
+
+    /**
+     * Convert change notification stream with buffering sentinels into stream of lists, where each
+     * list element contains a batch of data delineated by the markers.
+     *
+     * @return observable of non-empty list objects
+     */
+    public static <T> Transformer<ChangeNotification<T>, List<ChangeNotification<T>>> buffers() {
```

Comment:
We do not emit empty lists, which is what we would have to do if we have consecutive buffer sentinels issued. This is documented in @return description, but I will make it more explicit.